PWGHF/TableProducer/candidateCreatorLb.cxx

```diff
@@ -316,16 +311,14 @@ struct HfCandidateCreatorLbMc {
       rowMcMatchGen(flag, origin);
     }
   }
+  PROCESS_SWITCH(HfCandidateCreatorLbMc, processMc, "Process MC", false);
 };
 
 WorkflowSpec defineDataProcessing(ConfigContext const& cfgc)
 {
   WorkflowSpec workflow{
     adaptAnalysisTask<HfCandidateCreatorLb>(cfgc),
-    adaptAnalysisTask<HfCandidateCreatorLbExpressions>(cfgc)};
-  const bool doMC = cfgc.options().get<bool>("doMC");
-  if (doMC) {
-    workflow.push_back(adaptAnalysisTask<HfCandidateCreatorLbMc>(cfgc));
-  }
+    adaptAnalysisTask<HfCandidateCreatorLbExpressions>(cfgc),
+    adaptAnalysisTask<HfCandidateCreatorLbMc>(cfgc)};
```

Comment:
I tried that at first, but it does not work, because the process function of the MC task takes the table "HfCandLb", which is defined to be the extended table. So it depends on the extended task and I got an error. I can disentangle this more once I update the MC part. But right now I am only working on the data side.

Reply:
The extended table can be consumed in the same task that spawns it. As I said, that is how it is done in the other candidate creators. Is there a reason why it cannot work in your case?

Reply:
I get "[FATAL] error while setting up workflow in /Users/martinvoelkl/alice/sw/BUILD/df6e47ea8bea6924f26a4356915c5e1782d2308b/O2Physics/stage/bin/o2-analysis-hf-candidate-creator-lb: hf-candidate-creator-lb-expressions depends on itself". I assumed the dependency on the own table was the problem. What does it mean?

Reply:
I don't know. Can you push the code so that I can see?

Reply:
I see it now. The problem is the subscription `void processMc(aod::HfCandLb const& candidates,`.

Reply:
Thanks. See my comment above about the reason of the error.
Please do not force-push when your PR is under review.

Reply:
> I see it now. The problem is the subscription `void processMc(aod::HfCandLb const& candidates,`.

Yes. That is what I meant. That is why I had to put the process_switch you asked for in an MC task.